framework/src/org/checkerframework/framework/util/FieldInvariantObject.java

```diff
@@ -13,7 +13,9 @@
  * this field invariant applies.
  */
 public class FieldInvariantObject {
+    /** simple name of the field */
     private final List<String> fields;
+
```

Comment:
What qualifiers are these? Do all these qualifiers apply to all the fields? Or do the two lists have to be the same length and align? Instead of Lists should both fields be Sets?
The class javadoc talks about a `A list of fields that have a specific qualifier`, so it's not clear what this List represents.